logs-shipper/src/main/java/com/Resource.java

```diff
@@ -43,4 +64,81 @@ public String sendLog() {
         return response;
     }
 
+    @POST
+    @Path("index")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createTrackInJSON(RequestIndexMessage requestIndexMessage, @HeaderParam("user-agent") String userAgent) {
+        IndexRequest request = new IndexRequest("posts", "_doc");
+        request.id("" + ID++);
+
+        String msg = requestIndexMessage.getMessage();
+        if (msg == null) {
+            return Response.status(HttpURLConnection.HTTP_BAD_REQUEST).entity("The given object is null").build();
+        }
+
+        Map<String, String> jsonAsMap = new HashMap<>();
+        jsonAsMap.put("message", msg);
+        jsonAsMap.put("User-Agent", userAgent);
+
+        request.source(jsonAsMap);
+
+        client.indexAsync(request, RequestOptions.DEFAULT, new ActionListener<IndexResponse>() {
```

Comment:
If you use index async, how can you tell the client if his request succeeded?

Reply:
hmmm.....